source/priority-queue.ts

```diff
@@ -32,6 +33,31 @@ export default class PriorityQueue implements Queue<RunFunction, PriorityQueueOp
 		this.#queue.splice(index, 0, element);
 	}
 
+	setPriority(id: string, priority: number) {
+		const existingIndex: number = this.#queue.findIndex((element: Readonly<PriorityQueueOptions>) => element.id === id);
+		if (existingIndex === -1) {
+			throw new Error('Invalid Index - No promise function of specified id available in the queue.');
+		}
+
+		const [item] = this.#queue.splice(existingIndex, 1);
+		if (item === undefined) {
+			return;
```

Comment:
Is this state reachable?

Reply:
As per the existing linting settings, compiler suggests an error - of possible undefined value. Added a throw Error statement for the same.